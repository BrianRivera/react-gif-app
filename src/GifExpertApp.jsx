import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = props => {


    // const categorias = ['one punch man', 'Samurai x', 'Dragon Ball']
    const [categorias, setCategorias] = useState(['one punch man']);

    // const handleAdd = ()=> {
    //     setCategorias([...categorias,'naruto']);

    //importante, al declarar la priumera variable del set category se trae los datos del category
    //     setCategorias(cats =>[...cats,'naruto']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
            setCategorias={setCategorias}
             />
            <hr/>

        <ol>
            {
                categorias.map((resp) =>(
                <GifGrid 
                key={resp} 
                category={resp}>
                    {resp}
                </GifGrid>
                ))
            }
        </ol>
        </>
    )
}

export default GifExpertApp



/*
rafc = crear base completa
rafcp = crear base completa + proptyle
*/
