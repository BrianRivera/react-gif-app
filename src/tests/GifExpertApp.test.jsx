import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en GifExpertApp', () => {
const  wrapper = shallow(<GifExpertApp></GifExpertApp>);   

test('match con GifExpertApp', () => {
    expect(wrapper).toMatchSnapshot();
});

test('deve mostrar una lista de categorias', () => {
    const categories = ['sasha grey', 'stoya']
    const  wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>);  
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
});



})
