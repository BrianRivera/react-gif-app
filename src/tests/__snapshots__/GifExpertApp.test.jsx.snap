// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp deve mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="sasha grey"
      key="sasha grey"
    >
      sasha grey
    </GifGrid>
    <GifGrid
      category="stoya"
      key="stoya"
    >
      stoya
    </GifGrid>
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp match con GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch man"
      key="one punch man"
    >
      one punch man
    </GifGrid>
  </ol>
</Fragment>
`;
